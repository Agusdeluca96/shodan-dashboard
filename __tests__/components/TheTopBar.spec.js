import Vue from "vue";
import Vuex from "vuex";
import Vuetify from "vuetify";
import { createLocalVue, mount } from "@vue/test-utils";
import { user } from "@/store/user.module.js";
import { app } from "@/store/app.module.js";
import { UserMock } from "../__mocks__/user.mock";
import TheTopBar from "@/components/TheTopBar";

describe("TheTopBar.vue", () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  let vuetify;

  const userModuleActions = {
    signOut: jest.fn()
  };

  const appModuleActions = {
    toggleDrawerOpened: jest.fn(),
    setError: jest.fn()
  };

  const store = new Vuex.Store({
    modules: {
      app: {
        ...app,
        actions: appModuleActions
      },
      user: {
        ...user,
        actions: userModuleActions
      }
    }
  });
  store.commit("user/SET_CURRENT_USER", UserMock);

  const wrapperFactory = (wrapperProps = {}) => {
    return mount(TheTopBar, {
      store,
      localVue,
      vuetify,
      propsData: wrapperProps
    });
  };

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it("should match snapshot", () => {
    const wrapper = wrapperFactory();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should dispatch toggleDrawerOpened action when toggle drawer button is clicked", async () => {
    const wrapper = wrapperFactory();
    const toggleDrawerButton = wrapper.find(".v-app-bar__nav-icon");
    toggleDrawerButton.trigger("click");
    expect(appModuleActions.toggleDrawerOpened).toHaveBeenCalled();
  });

  it("should dispatch signOut action when signOut list item is clicked", async () => {
    const wrapper = wrapperFactory();
    // First open the menu to being able of finding the sign out list item
    const profileMenuButton = wrapper.find("#profile-menu-button");
    profileMenuButton.trigger("click");
    await Vue.nextTick();

    const signOutListItem = wrapper.find("#sign-out-list-item");
    signOutListItem.trigger("click");
    expect(userModuleActions.signOut).toHaveBeenCalled();
  });
});
