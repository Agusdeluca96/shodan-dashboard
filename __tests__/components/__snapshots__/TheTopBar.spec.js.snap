// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheTopBar.vue should match snapshot 1`] = `
<header class="v-sheet theme--light v-toolbar v-toolbar--absolute v-toolbar--flat v-app-bar transparent" style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;" id="app-top-bar">
  <div class="v-toolbar__content" style="height: 64px;"><button type="button" class="v-app-bar__nav-icon v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu theme--light"></i></span></button>
    <div class="spacer"></div>
    <div class="v-menu"><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default" id="profile-menu-button" role="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content"><div class="v-avatar" style="height: 32px; min-width: 32px; width: 32px;" item=""><div aria-label="Agustin De Luca" role="img" class="v-image v-responsive theme--light"><transition-stub name="fade-transition" mode="in-out"><div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div></transition-stub><div class="v-responsive__content"></div></div></div></span></button>
      <!---->
    </div>
  </div>
</header>
`;
